src/test/java/bio/terra/service/dataset/DatasetAzureIntegrationTest.java

```diff
@@ -395,6 +405,29 @@ public void datasetIngestFileHappyPath() throws Exception {
         dataRepoFixtures.ingestJsonData(steward, datasetId, ingestRequestCSV);
     assertThat("2 row were ingested", ingestResponseCSV.getRowCount(), equalTo(2L));
 
+    // Only check the subset of tables that have rows
+    Set<String> tablesToCheck = Set.of(jsonIngestTableName, ingest2TableName);
+    // Read the ingested metadata
+    AccessInfoParquetModel datasetParquetAccessInfo =
+        dataRepoFixtures
+            .getDataset(
+                steward(), datasetId, List.of(DatasetRequestAccessIncludeModel.ACCESS_INFORMATION))
+            .getAccessInformation()
+            .getParquet();
+
+    String datasetParquetUrl =
+        datasetParquetAccessInfo.getUrl() + "?" + datasetParquetAccessInfo.getSasToken();
+    TestUtils.verifyHttpAccess(datasetParquetUrl, Map.of());
+    verifySignedUrl(datasetParquetUrl, steward(), "rl");
+
+    for (AccessInfoParquetModelTable table : datasetParquetAccessInfo.getTables()) {
+      if (tablesToCheck.contains(table.getName())) {
+        String tableUrl = table.getUrl() + "?" + table.getSasToken();
```

Comment:
The use of `?` embeds the knowledge of how SAS token URLs are constructed. In case that format changes in the future, it might be nice to consolidate that knowledge in the class that builds and deconstructs them, like `UrlParts`.

Reply:
Maybe.  Though admittedly, this is strictly fr test code.  That's actually why I separated it out in the base code since it typically isn't used like this (just for the test, it was a bit easier to test)

Reply:
That we get a valid URL.  But I also want to know if this starts breaking if the format gets changed form under us